src/overlays/actors/ovl_Door_Shutter/z_door_shutter.c

```diff
@@ -798,7 +874,7 @@ void DoorShutter_GohmaBlockFall(DoorShutter* this, PlayState* play) {
         if (!GET_EVENTCHKINF(EVENTCHKINF_70)) {
             BossGoma* parent = (BossGoma*)this->dyna.actor.parent;
 
-            this->unk_164 = 10;
+            this->moveState = 10;
```

Comment:
So `moveState` is used both as a timer and as a bool to indicate in-use? Does it make sense for player to set the timer to 1 instead of a bool being set to true?

Reply:
Perhaps this would be a good use of a union, since this one variable does seem to have two different usages (1 as a boolean and 1 as a timer)?

Reply:
Note `moveState` is already sketchy from being from the `SLIDING_DOOR_ACTOR_BASE` macro

I'm not opposed to making a union but I don't think it's necessary

The name isn't completely inappropriate either, gohma block / phantom ganon bars use it as a movement-related timer, so "move state" isn't that bad (right? <img src="https://cdn.discordapp.com/emojis/982515528096682014.webp?size=240&quality=lossless" alt="COPIUM" height="24">)

Reply:
I will say I would probably prefer a union, more so the player usage with a boolean named `moveState` seems strange. Or perhaps just name the boolean more boolean like, and just comment the hacky nature for `gohma block / phantom ganon bars'.

(I will say that this isn't a deal breaker, so will approve anyway :) )

Reply:
the gohma case does seem more of an afterthought, would agree with naming it after its main purpose and commenting on its re-use

Reply:
Does "moveState" not work for the player interaction?

I was thinking of it as "when the player interacts with the door, get the door to move"
(and it's not far from gohmablock/phantomganonbars using it as a movement timer)

What would be a better name?

Reply:
I'd prefer something like `isUsing` or maybe `isMoving`.

The main thing about `moveState` is thinking of a `true` moveState just seems strange. It seems more of something that needs an enum associated with it, rather than acting as a boolean.

Reply:
-> `isActive`